src/App.js

```diff
@@ -1,24 +1,80 @@
-import logo from './logo.svg';
-import './App.css';
+import { useEffect, useState } from 'react'
+import BarList from './components/BarList'
+import './App.css'
 
 function App() {
+  const [barList, setBarList] = useState([])
+  const [isLoading, setIsLoading] = useState(false)
+  const [locationUnavailable, setLocationUnavailable] = useState(false)
+  const [currentLocation, setCurrentLocation] = useState({})
+  const [error, setError] = useState(null)
+  
+  const success = (position) => {
+    const { latitude, longitude } = position.coords
+    setCurrentLocation({ latitude, longitude })
+  }
+  
+  const failure = () => {
+    setLocationUnavailable(true)
+  }
+  
+  const getLocation = () => {
+    navigator.geolocation.getCurrentPosition(success, failure)
+  }
+  
+  useEffect(() => {
+    if ("geolocation" in navigator) {
+      getLocation()
+    } else {
+      setLocationUnavailable(true)
+    }
+  }, [])
+  
+  const findBarsHandler = async () => {
+    setIsLoading(true)
+    try {
+      const url = 'https://api.openbrewerydb.org/v1/breweries'
```

Comment:
Could this be better stored in the state or some globally accessible location as it will probably be used in multiple locations (search, pagination, etc)?